File: src/main/java/pe/tarjetaestilos/models/FormBlog.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;
import java.time.ZonedDateTime;


@Entity
@Table(name="form_blog")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FormBlog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(columnDefinition = "TEXT")
    private String dni;
    
    @Column(columnDefinition = "TEXT")
    private String phone;
    
    @Column(columnDefinition = "TEXT")
    private String email;
    
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;
    
    
    @ManyToOne()
    @JoinColumn(name = "menu_id", nullable = false)
    @ToString.Exclude
    private Menu menu;
}

File: TarjetaEstilosVueDasboard/src/models/FormBlog.ts
class FormBlog{
    dni:string|null;
    phone:string|null;
    email:string|null;
    accept:boolean|null;
    created_at:Date|null;
    id:number|null;
    menu_id:number|null;

    constructor(
        dni:string|null,
        phone:string|null,
        email:string|null,
        accept:boolean|null,
        created_at:Date|null,
        menu_id:number|null=null,
        id:number|null
    ){
        this.dni = dni;
        this.phone = phone;
        this.email = email;
        this.accept = accept;
        this.created_at = created_at;
        this.id = id;
        this.menu_id = menu_id;
    }
};
export default FormBlog;